br/pkg/storage/s3.go

```diff
@@ -245,6 +245,11 @@ func newS3Storage(backend *backuppb.S3, opts *ExternalStorageOptions) (*S3Storag
 		WithS3ForcePathStyle(qs.ForcePathStyle).
 		WithRegion(qs.Region)
 	request.WithRetryer(awsConfig, defaultS3Retryer())
+
+	if len(qs.Endpoint) > 0 {
+		strings.TrimSuffix(qs.Endpoint, "/")
```

Comment:
Strings in golang are immutable, functions in `strings` always returns a new string(instead of change it in place).🤔

```suggestion
		qs.Endpoint = strings.TrimSuffix(qs.Endpoint, "/")
```

Reply:
fixed